lib/ecto/changeset.ex

```diff
@@ -1217,6 +1217,19 @@ defmodule Ecto.Changeset do
     raise ArgumentError, "field names given to change/put_change must be atoms, got: `#{inspect(key)}`"
   end
 
+  defp put_change(data, changes, errors, valid?, key, new_value, {:parameterized, type, params}) do
+    old_value = Map.get(data, key)
+
+    if type.equal?(old_value, new_value, params) in [:skip, false] do
```

Comment:
Sorry, I mean to say that `type.change` will return `:skip`, not `type.equal?`. :) We would skip this call altogether!

Reply:
done